pkg/healthcheck/healthcheck.go

```diff
@@ -1450,9 +1450,12 @@ func CheckProxyVersionsUpToDate(pods []corev1.Pod, versions version.Channels) er
 // matches that of the CLI
 func CheckIfProxyVersionsMatchWithCLI(pods []corev1.Pod) error {
 	for _, pod := range pods {
-		proxyVersion := k8s.GetProxyVersion(pod)
-		if proxyVersion != "" && proxyVersion != version.Version {
-			return fmt.Errorf("%s running %s but cli running %s", pod.Name, proxyVersion, version.Version)
+		status := k8s.GetPodStatus(pod)
+		if status == string(corev1.PodRunning) && containsProxy(pod) {
+			proxyVersion := k8s.GetProxyVersion(pod)
```

Comment:
`GetProxyVersion` will return an empty string if the pod doesn't contain a proxy . So there's no need to call `containsProxy`. You can put all the conditions (non-empty version, version comparison and status) in the same `if` statement.